sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -183,9 +183,13 @@ class __SYCL_EXPORT kernel_bundle_plain {
 
   bool native_specialization_constant() const noexcept;
 
-  bool ext_oneapi_has_kernel(const std::string &name);
+  bool ext_oneapi_has_kernel(const std::string &name) {
+    return ext_oneapi_has_kernel(detail::string_view{name});
```

Comment:
`std::string_view`?

Reply:
No, detail::string_view is correct.
I introduced detail::string_view to handle ABI-incompatibility issue when the user compiles their code with -D_GLIBCXX_USE_CXX11_ABI=0.
To make story short, std::string and std::string_view cannot be used when there is an ABI-compatibility issue.

Reply:
Can you please add this comment to the declaration of `ext_oneapi_has_kernel(detail::string_view)`? And maybe also to some generic doc

Reply:
I have been doing these changes in all over the places of SYCL.
I already added similar comments in detail::string and detail::string_view.
https://github.com/intel/llvm/blob/60be73405ef228b89a5886fcbe6abfd18022c4dd/sycl/include/sycl/detail/string.hpp#L18
So, that should cover all the changed like this PR.